### PR TITLE
Added Total and Average score for FeltStigma and PSS subforms

### PR DESCRIPTION
**Pull request by ghulamjilani**

### CODE REVIEW

File: app/services/record_json_validator_service.rb [hunk 1]
```diff
@@ -51,6 +51,8 @@ def build_schema(fields)
         properties[field.name] = { 'type' => %w[string null] }
       when Field::TALLY_FIELD
         properties[field.name] = { 'type' => %w[object null], 'properties' => tally_properties(field.tally_i18n) }
+      when Field::CALCULATED
```
[the remark this answers is not in the document]
Comment (ghulamjilani) [reply app/services/record_json_validator_service.rb:54]: Only numeric data is passed. I think we don't need to add other types.
Please explain this if we need other types as well.

File: app/javascript/components/record-form/form/utils/get-values-count.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { isNil, has } from "lodash";
```
Comment (jtoliver-quoin) [inline app/javascript/components/record-form/form/utils/get-values-count.js:1]: Seperate lodash imports example...

`import isNil from "lodash/isNil"`

File: app/services/record_json_validator_service.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (pnabutovsky) [reply app/services/record_json_validator_service.rb:54]: Calculated fields should be display-only fields according to a previous conversation. They can certainly have a numeric  value (derived from the calculations) but they can be strings (imagine a concat function being defined at a latter point) and booleans as well (see existing boolean logic).

File: app/javascript/components/record-form/form/subforms/subform-header/component.jsx [hunk 3]
```diff
@@ -83,6 +83,7 @@ const Component = ({ field, values, locale, displayName, index, isViolationSubfo
 
     return (
       <ListItemText id="subform-header-button" classes={itemClasses}>
+        {values[index][`${field.name.split("_section")[0]}_date`]}
```
Comment (dhernandez-quoin) [inline app/javascript/components/record-form/form/subforms/subform-header/component.jsx:86]: This probably is not the solution for this issue. Can you confirm that your configuration has: `collapsed_field_names: ['felt_stigma_date']` and that after loading the configuration this becomes: collapsed_field_names: ['felt_stigma_scale']

Comment (ghulamjilani) [reply app/javascript/components/record-form/form/subforms/subform-header/component.jsx]: I have removed this code. I think someone has fixed the issue. Collapsed field is working for me.
Screenshot 2022-03-17 14:42:05

**Review by pnabutovsky — APPROVED** [review]

**Pull request merged**